### ISSUE TITLE
Upgrading from [10.2.0] => [10.3.0] add a breaking change on field name case sensitivity

### ISSUE THREAD

**Issue by erwan-joly** [ChilliCream/hotchocolate#2617]
**Describe the bug**
We have been renaming a field in the past as it's casing was not following our standards. It used to works fine until the upgrade to 10.3.0 that stopped having a case sensitivity on the field keys.

**To Reproduce**
1. Create an input type with two similar fields but different casing
```
		public string YourFieldName { get; set; }
		[GraphQLDeprecated("This is deprecated")]
		public string YourFieldname{ get; set; }
```
2. Build and Run

_with v10.3.0_
3. See ```System.ArgumentException: 'An item with the same key has already been added. Key: YourFieldName'```
_with v10.2.0_
3. Nothing to see it's working fine

**Expected behavior**
I would expect two different casing to be valid and not counted as same Key

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 10

**Comment by michaelstaib** [comment]
This might be an issue with 10.3... We have released 10.5.4 yesterday, so your issue might be fixed.

**Comment by michaelstaib** [comment]
Also, 10.5.4 is the last fix for 10.x. we are releasing 11 on Monday and will focus only on 11 going forward. I will close this issue.

**Comment by michaelstaib** [comment]
Just checked and added a test to 11:

```
type QueryFieldCasing {
  yourFieldName: String
  yourFieldname: String @deprecated(reason: "This is deprecated")
}
```

So, this should also be fixed with either the last 10.3.x release or the last 10.5.x release.

**Comment by erwan-joly** [comment]
@michaelstaib 
it seems like it only happen on input types.
Since 10.3.0 and still here in 11.0.0-rc.6
```
using HotChocolate;
using HotChocolate.Types;
using Microsoft.AspNetCore;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.DependencyInjection;

namespace ProofOfBug
{
    class Program
    {
        public static void Main(string[] args)
        {
            CreateWebHostBuilder(args).Build().Run();
        }

        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseUrls("http://localhost:5050")
                .UseStartup<Startup>();
    }
    public class Startup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddGraphQL(Schema.Create(c =>
            {
                c.RegisterQueryType<QueryType>();
            }));
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            app.UseGraphQL();
        }
    }
    public class Query
    {
        public string DoFoo(Foo foo)
        {
            return "foo";
        }
    }

    public class QueryType
        : ObjectType<Query>
    {
        protected override void Configure(
            IObjectTypeDescriptor<Query> descriptor)
        {
            descriptor.Field(t => t.DoFoo(default))
                .Argument("foo", a => a.Type<FooType>())
                .Type<StringType>();
        }
    }

    public class FooType
        : InputObjectType<Foo>
    {
        
    }
    public class Foo
    {
        public string YourFieldName { get; set; }
        [GraphQLDeprecated("This is deprecated")]
        public string YourFieldname { get; set; }
    }
}
```

**Comment by michaelstaib** [comment]
Let me check

**Comment by PascalSenn** [comment]
@michaelstaib  maybe this fixes it:
https://github.com/ChilliCream/hotchocolate/pull/2630

**Comment by michaelstaib** [comment]
We have postponed the fix and will deliver it with the first patch build of 11.

**Comment by michaelstaib** [comment]
This is now finally fixed with 12.0.0-rc.8
